packages/common/src/sites/feeds/previewFeed.ts

```diff
@@ -0,0 +1,51 @@
+import { IArcGISContext } from "../../ArcGISContext";
+import { FeedFormat } from "./types";
+
+export interface IPreviewFeedOptions {
+  format: FeedFormat;
+  version: string;
+  previewTemplate: Record<string, any>;
+  previewHubId: string;
+  context: IArcGISContext;
+}
+
+/**
+ * Preview a feed using the provided template and context
+ * @param opts.format - the feed format
+ * @param opts.version - the feed version
+ * @param opts.previewTemplate - the feed template to preview
+ * @param opts.previewHubId - the Hub ID to preview the feed for
+ * @param opts.context - the ArcGIS context
+ * @returns the previewed feed as a string
+ */
+export async function previewFeed(opts: IPreviewFeedOptions): Promise<string> {
+  const { format, version, previewHubId, context } = opts;
+  const stringifiedTemplate = JSON.stringify(opts.previewTemplate);
+
+  const baseUrl = `${context.hubUrl}/api/feed/${format}/${version}`;
```

Comment:
I've made a mistake in the past using a URL that could occasionally include a different base route for the API or include query parameters. That can't happen here, can it?

Reply:
Searching hub.js I see `getHubApiUrl`, `getHubApiUrlFromPortal`, and `getHubApiUrlRoot`

Reply:
@abp6318 excellent question! I talked with @tomwayson and we are good to use `context.hubUrl` in this scenario. All the functions you mentioned above pre-date `IArcGISContext`, so we don't want to rely on them if we don't have to.